pkg/stargate/deployments.go

```diff
@@ -50,19 +50,16 @@ const (
 
 var (
 	defaultImage3 = images.Image{
-		Registry:   images.DefaultRegistry,
```

Comment:
[issue] I agree that we can remove this, but there is now a problem with the way we handle defaults.

I've created a test environment and manually configured the operator deployment with a wrong registry value:
```yaml
- name: DEFAULT_REGISTRY
  value: localhost:1234
```
Then created a cluster with a Stargate spec that does not specify the image. I would expect this to fail, but it doesn't. It turns out the operator is still pulling Stargate from docker.io.

I think the problem comes from the fact that the `Image` CRD provides a default value for the field:
```go
	// +kubebuilder:default="docker.io"
	// +optional
	Registry string `json:"registry,omitempty"`
```
Therefore when we hit this instruction in `ApplyDefaults()` (same source file)...
```go
	if merged.Registry == "" {
		merged.Registry = DefaultRegistry
	}
```
... `merged.Registry` is "docker.io", and the default is not applied.

I think the solution is to remove the default value in the CRD. I've validated that it solves the problem in my test.